remove accelerator_view::get_hsa_queue()

ssahasra opened the pull request:
This function returns the underlying hsa_queue_t for an
accelerator_view, but this mapping is not stable in the presence of
queue stealing. It was introduced for HIP, but it is not used
anywhere, and it is unsafe anyway.

scchan reviewed (changes requested):
Could you remove this test?

Failing Tests (1):
    HCC :: Unit/HC/test1.cpp

ssahasra committed:
remove accelerator_view::get_hsa_queue()

This function returns the underlying hsa_queue_t for an
accelerator_view, but this mapping is not stable in the presence of
queue stealing. It was introduced for HIP, but it is not used
anywhere, and it is unsafe anyway.

ssahasra commented:
> Could you remove this test?
> 
> Failing Tests (1):
> HCC :: Unit/HC/test1.cpp

Done.
